functorch/functorch/_src/aot_autograd.py

```diff
@@ -237,6 +237,13 @@ def fake_fn(primals, tangents):
 
                 with track_graph_compiling("joint"):
                     fw_module, bw_module = partition_fn(fx_g, joint_inputs)
+                    # If the original user's program mutated any tensor inputs,
+                    # it should show up in the forward module.
+                    # When functinalization is turned on, we want to move
+                    # these mutations into an opaque submodule
+                    # so our graph infra can assume a functional graph.
+                    if config.use_functionalize:
+                        move_input_mutations_into_submodule(fw_module)
```

Comment:
So if you do this after, `partition_fn` must be able to deal with mutations in `fx_g`. Can it, @Chillee ?

Reply:
I re-remembered that it can't haha, so I'll have to change this. the min-cut partition function code calls `eliminate_dead_code()`, so we want the `copy_()` ops to be hidden away before then.